src/logic/Board.ts

```diff
@@ -108,13 +103,31 @@ export class Board implements IBoard {
   }
 
   /**
-   * checks if a row has space to accommodate a pushed piece
-   * @param row a row of Tiles
-   * @returns boolean
+   * Checks if the row can accommodate a new tile.
```

Comment:
we need to be careful about the distinction between a Tile and a game piece - we're not really pushing a new tile in, just the fill